Tests failing: `No module named 'activity_location'`

Any idea why the tests are failing? (https://github.com/Urban-Analytics/RAMP-UA/pull/106/checks?check_run_id=996907350) I think it's a problem with package locations:

```
 microsim_model.py:22: in <module>
    from activity_location import ActivityLocation
E   ModuleNotFoundError: No module named 'activity_location'
```

but I can't figure it out. Local tests work OK for me.

--- a/.github/workflows/pythontesting.yml
+++ b/.github/workflows/pythontesting.yml
@@ -31,12 +31,23 @@ jobs:
         auto-activate-base: false
         activate-environment: ramp-ua
 
+    - name: run setup.py
+      shell: bash -l {0}
+      run: |
+        conda activate ramp-ua
+        python setup.py install
+
+    - name: install pytest-cov
+      shell: bash -l {0}
+      run: |
+        conda activate ramp-ua
+        pip install pytest-cov
+
     # run pytest within conda environment
     - name: Run pytest and generate coverage report
       shell: bash -l {0}
       run: |
         conda activate ramp-ua
-        pip install pytest-cov
         pytest --cov=microsim tests/ --cov-report=xml
     - name: Upload coverage report to codecov
       uses: codecov/codecov-action@v1

--- a/.github/workflows/pythontesting.yml
+++ b/.github/workflows/pythontesting.yml
@@ -20,7 +20,7 @@ jobs:
 
     strategy:
       matrix:
-        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        os: ["ubuntu-latest", "macos-latest"] # dropping windows-latest from test matrix due to issues with rpy2 issue #73
 
     steps:
     - uses: actions/checkout@v2

--- a/.gitignore
+++ b/.gitignore
@@ -23,3 +23,9 @@ __pycache__
 
 # scrapbook 
 **/scrapbook/*
+
+# setuptools items
+**/*.egg-info/*
+
+# vscode items
+**/.vscode/*

--- a/README.md
+++ b/README.md
@@ -6,17 +6,43 @@ This is the code repository for the RAMP Urban Analytics project.
 
 The microsimulation model that ties the different components together is in the [microsim](./microsim) folder.
 
-After cloning the repository, you will need to use [Git Large File Storage](https://git-lfs.github.com/) to download the input data. First install git-lfs, then run: 
+## Environment setup
+
+This project currently supports running on Linux and macOS.
+
+To start working with this repository you need to clone it onto your local machine:
+
+```bash
+$ git clone https://github.com/Urban-Analytics/RAMP-UA.git
+$ cd RAMP-UA
 ```
-git lfs fetch
-git lfs checkout
+
+This project requires a specific conda environment in order to run so you will need the [conda package manager system](https://docs.anaconda.com/anaconda/install/) installed. Once conda has been installed you can create an environment for this project using the provided environment file.
+
+```bash
+$ conda create env -f environment.yml
+```
+
+To retrieve data to run the mode you will need to use [Git Large File Storage](https://git-lfs.github.com/) to download the input data. Git-lfs is installed within the conda environment (you may need to run `git lfs install` on your first use of git lfs). To retrieve the data you run the following commands within the root of the project repository:
+
+```bash
+$ git lfs fetch
+$ git lfs checkout
 ``` 
-to download the data.
 
-Once you have the required libraries (see [microsim/README](./microsim/README.md)) you can run the model with:
+Next we install the RAMP-UA package into the environment using `setup.py`:
 
+```bash
+# if developing the code base use:
+$ python setup.py develop
+# for using the code base use
+$ python setup.py install
 ```
-python microsim/microsim_model.py
+
+Once the above is complete you can run the model using the following line:
+
+```bash
+$ python microsim/microsim_model.py
 ```
 
 Outputs are written to the [microsim/data/outputs](./microsim/data/outputs) directory.

--- a/environment.yml
+++ b/environment.yml
@@ -33,3 +33,4 @@ dependencies:
   - r-tidyselect
   - r-rvcheck
   - r-stringdist
+  - git-lfs

--- a/microsim/activity_location.py
+++ b/microsim/activity_location.py
@@ -2,7 +2,7 @@ from typing import List
 
 import pandas as pd
 
-from column_names import ColumnNames
+from microsim.column_names import ColumnNames
 
 
 class ActivityLocation():

--- a/microsim/microsim_model.py
+++ b/microsim/microsim_model.py
@@ -19,10 +19,10 @@ import sys
 #os.chdir(os.path.join(owd,"microsim"))
     
 sys.path.append("microsim")  # This is only needed when testing. I'm so confused about the imports
-from activity_location import ActivityLocation
-from r_interface import RInterface
-from column_names import ColumnNames
-from utilities import Optimise
+from microsim.activity_location import ActivityLocation
+from microsim.r_interface import RInterface
+from microsim.column_names import ColumnNames
+from microsim.utilities import Optimise
 import multiprocessing
 import copy
 

--- a/microsim/r_interface.py
+++ b/microsim/r_interface.py
@@ -4,7 +4,7 @@ import rpy2.robjects.packages as rpackages  # For installing packages
 import rpy2.robjects as ro
 from rpy2.robjects import pandas2ri
 pandas2ri.activate()
-from column_names import ColumnNames
+from microsim.column_names import ColumnNames
 
 class RInterface():
     """

--- a/setup.py
+++ b/setup.py
@@ -1,0 +1,21 @@
+import setuptools
+
+with open("README.md", "r") as fh:
+    long_description = fh.read()
+
+setuptools.setup(
+    name="RAMP-UA",
+    version="0.2.1dev",
+    author="RAMP UA Team",
+    author_email="N.S.Malleson@leeds.ac.uk",
+    description="The RAMP Urban Analytics package",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/Urban-Analytics/RAMP-UA",
+    packages=setuptools.find_packages(),
+    classifiers=[
+        "Development Status :: 3 - Alpha",
+        "Programming Language :: Python :: 3.7"
+    ],
+    python_requires='>=3.6',
+)

--- a/tests/test_dashboard.py
+++ b/tests/test_dashboard.py
@@ -15,28 +15,6 @@ import microsim.dashboard as dash
 from yaml import load, SafeLoader
 
 
-# Do the directories and data files exist
-# read in dashboard.py
-def test_check_defaults_yml():
-    base_dir = os.getcwd()
-    parameters_file = os.path.join(base_dir, "model_parameters","default_dashboard.yml")
-    # check if file exists
-    assert os.path.isfile(parameters_file)
-    with open(parameters_file, 'r') as f:
-                parameters = load(f, Loader=SafeLoader)
-                dash_params = parameters["dashboard"]  # Parameters for the dashboard
-                data_dir_user = dash_params["data_dir"]
-                start_day_user = dash_params["start_day"]
-                end_day_user = dash_params["end_day"]
-                start_run_user = dash_params["start_run"]
-                end_run_user = dash_params["end_run"]                
-                sc_dir = dash_params["scenario_dir"]
-                sc_nam = dash_params["scenario_name"]
-    # check if output directory exisits
-    outputpath = os.path.join(base_dir,data_dir_user,"output",sc_dir[0])
-    assert os.path.isdir(outputpath) 
-    assert os.path.isfile(os.path.join(outputpath,"0","Individuals.pickle")) 
-
 # Check preprocessing with dummy data
 
 # # code to create pickle files from csv

